### DISCUSSION TITLE
Customise Caddyfile / Expanded default Caddyfile

### DISCUSSION THREAD

**Discussion by nikspyratos** [deployphp/deployer#3666] [Help needed]
Hi there,

Is there a recommended way of having a custom Caddyfile for a provisioned site? Currently the template is hardcoded in the `provision:website` task. What I'm thinking of is using the same command used [here](https://github.com/deployphp/deployer/pull/2949/files#diff-ffb7be82353447e226fbe65ce4a0a60a3b70eb7dfe131ce0f30845bd4497c87cR73) to overwrite the Caddyfile with an expanded one based on the default.

Alternatively, Is there any particular objection to expanding this template? I think there are some sane defaults [here](https://ma.ttias.be/caddyfile-config-example-for-laravel/) and [here](https://jorgeglz.io/blog/setting-up-laravel-applications-with-caddy-2/) that apply to most PHP applications and not just Laravel - `file_server` directive, zstd and gzip encoding, the safety rewrites and default headers.

**Comment by antonmedv** [comment]
Sure. Let's improve the template.

**Reply by nikspyratos** [reply]
Will make a PR soon!

**Reply by nikspyratos** [reply]
#3669
